staging/src/k8s.io/apiserver/pkg/util/proxy/proxy.go

```diff
@@ -114,12 +109,9 @@ func ResolveCluster(services listersv1.ServiceLister, namespace, id string) (*ur
 			Host:   net.JoinHostPort(svc.Spec.ClusterIP, fmt.Sprintf("%d", svcPort.Port)),
 		}, nil
 	case svc.Spec.Type == v1.ServiceTypeExternalName:
-		if port.Type != intstr.Int {
-			return nil, fmt.Errorf("named ports not supported")
-		}
 		return &url.URL{
 			Scheme: "https",
-			Host:   net.JoinHostPort(svc.Spec.ExternalName, port.String()),
+			Host:   net.JoinHostPort(svc.Spec.ExternalName, string(port)),
```

Comment:
casting an int port to a string doesn't do what you think it does - https://play.golang.org/p/4teKAho_jyr :)

need to do `strconv.Itoa()`

Reply:
interesting :)